mindmeld/components/entity_resolver.py

```diff
@@ -51,51 +65,189 @@ class EntityResolver:
+        If supported, override this method definition in the derived class
+        (eg. see EntityResolverUsingElasticSearch)
+        """
+        logger.warning(
+            "%r not configured to use double_metaphone",
+            self.name
+        )
+        raise NotImplementedError
+
+    def cache_path(self, tail_name=""):
+        name = self.name + "_" + tail_name if tail_name else self.name
+        return path.get_entity_resolver_cache_file_path(
+            self.app_path, self.type, name
+        )
+
+    @abstractmethod
+    def _fit(self, clean):
+        raise NotImplementedError
+
+    def fit(self, clean=False):
+        """Fits the resolver model, if required
+
+        Args:
+            clean (bool, optional): If ``True``, deletes and recreates the index from scratch
+                                    with synonyms in the mapping.json.
+        """
+
+        if self.ready:
+            return
+
+        if self._no_canonical_entity_map:
+            return
+
+        self._fit(clean)
+        self.ready = True
+
+    @abstractmethod
+    def _predict(self, entity):
+        raise NotImplementedError
+
+    def predict(self, entity):
+        """Predicts the resolved value(s) for the given entity using the loaded entity map or the
+        trained entity resolution model.
+
+        Args:
+            entity (Entity, tuple): An entity found in an input query, or a list of n-best entity \
+                objects.
+
+        Returns:
+            (list): The top 20 resolved values for the provided entity.
+        """
+        if isinstance(entity, (list, tuple)):
+            top_entity = entity[0]
+            entity = tuple(entity)
+        else:
+            top_entity = entity
+            entity = tuple([entity])
+
+        if self._is_system_entity:
+            # system entities are already resolved
+            return [top_entity.value]
+
+        if self._no_canonical_entity_map:
+            return []
+
+        return self._predict(entity)
+
+    @abstractmethod
+    def _load(self):
+        raise NotImplementedError
+
+    def load(self):
+        """If available, loads embeddings of synonyms that are previously dumped
+        """
+        self._load()
+
+    def _dump(self):
+        raise NotImplementedError
+
+    def __repr__(self):
+        msg = "<{} {!r} ready: {!r}, dirty: {!r}>"
+        return msg.format(self.__class__.__name__, self.name, self.ready, self.dirty)
+
+
+class EntityResolverUsingElasticSearch(EntityResolverBase):
+    """
+    Resolver class based on Elastic Search
+    """
+
+    # prefix for Elasticsearch indices used to store synonyms for entity resolution
+    ES_SYNONYM_INDEX_PREFIX = "synonym"
+    """The prefix of the ES index."""
+
+    def __init__(self, app_path, resource_loader, entity_type, er_config, **kwargs):
+        super().__init__(app_path, resource_loader, entity_type, er_config, **kwargs)
+        self._es_host = self.kwargs.get("es_host", None)
```

Comment:
Its more readable to access this parameter from the function kwargs ie:
```python
def __init__(self, app_path, resource_loader, entity_type, er_config, es_host=None, **kwargs):
    ....
    self._es_host = es_host
```

Reply:
I felt that sending different set of arguments for different derived classes (based on the type of algorithm) isn't a good idea, as it will add an additional _if-else_ condition in the base class, right? I think object creation can be kept generic. What do you think?